src/libstd/sys_common/backtrace.rs

```diff
@@ -69,16 +69,18 @@ fn _print(w: &mut Write, format: PrintFormat) -> io::Result<()> {
         inline_context: 0,
     }; MAX_NB_FRAMES];
     let (nb_frames, context) = unwind_backtrace(&mut frames)?;
-    let (skipped_before, skipped_after) =
-        filter_frames(&frames[..nb_frames], format, &context);
-    if skipped_before + skipped_after > 0 {
+    let filtered_frames = filter_frames(&frames[..nb_frames], &context, format);
+    if filtered_frames.len() != nb_frames {
         writeln!(w, "note: Some details are omitted, \
                      run with `RUST_BACKTRACE=full` for a verbose backtrace.")?;
     }
     writeln!(w, "stack backtrace:")?;
 
-    let filtered_frames = &frames[..nb_frames - skipped_after];
-    for (index, frame) in filtered_frames.iter().skip(skipped_before).enumerate() {
+    for (index, frame, is_on_filter_edge) in filtered_frames {
+        // Don't use ANSI escape codes on windows, because most terminals on it don't support them
+        if is_on_filter_edge && cfg!(not(windows)) {
+            write!(w, "\x1B[2m")?;
```

Comment:
Dealing with ansi codes can be sort of a minefield unfortunately. This isn't handling terminals on Unix, for example, that don't support ansi codes. It also doesn't handle cases on Windows where ansi codes are supported (like MSYS and newer Windows terminals I think).

I'm not entirely certain what this ansi code is doing, but could it be left out for now?

Reply:
This ansi code makes the line darker, as it is less important.

Reply:
Ok, in that case can this be omitted for now? I'm not sure that libstd is really prepared to accurately place ANSI codes everywhere